client/src/redux/sagas/datafiles.sagas.js

```diff
@@ -756,3 +756,180 @@ export function* trashFile(system, path, id) {
     });
   }
 }
+
+export const getLatestApp = async name => {
+  const res = await fetchUtil({
+    url: '/api/workspace/apps',
+    params: { publicOnly: true }
+  });
+  const apps = res.response;
+
+  const latestApp = apps
+    .filter(app => app.id.includes(name))
+    .reduce(
+      (latest, app) => {
+        if (app.version > latest.version) {
+          return app;
+        }
+        if (app.version < latest.version) {
+          return latest;
+        }
+        // Same version of app
+        if (app.revision >= latest.revision) {
+          return app;
+        }
+        return latest;
+      },
+      { revision: null, version: null }
+    );
+  return latestApp.id;
+};
+
+const getExtractParams = (file, latestExtract) => {
+  const inputFile = `agave://${file.system}${file.path}`;
+  const archivePath = `agave://${file.system}${file.path.substring(
+    0,
+    file.path.lastIndexOf('/') + 1
+  )}`;
+  return JSON.stringify({
+    allocation: 'FORK',
+    appId: latestExtract,
+    archive: true,
+    archivePath,
+    inputs: {
+      inputFile
+    },
+    maxRunTime: '02:00:00',
+    name: 'Extracting Compressed File',
+    parameters: {}
+  });
+};
+
+export function* extractFiles(action) {
+  try {
+    const latestExtract = yield call(getLatestApp, 'extract-frontera');
+    const params = getExtractParams(action.payload.file, latestExtract);
+    yield put({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: 'RUNNING', operation: 'extract' }
+    });
+    const submission = yield call(jobHelper, params);
+    if (submission.execSys) {
+      // If the execution system requires pushing keys, then
+      // bring up the modal and retry the extract action
+      yield put({
+        type: 'SYSTEMS_TOGGLE_MODAL',
+        payload: {
+          operation: 'pushKeys',
+          props: {
+            onSuccess: action,
+            system: submission.execSys,
+            onCancel: {
+              type: 'DATA_FILES_SET_OPERATION_STATUS',
+              payload: { status: 'ERROR', operation: 'extract' }
+            }
+          }
+        }
+      });
+    } else if (submission.status === 'ACCEPTED') {
+      yield put({
+        type: 'DATA_FILES_SET_OPERATION_STATUS',
+        payload: { status: 'SUCCESS', operation: 'extract' }
+      });
+    } else {
+      throw new Error('Unable to extract files');
+    }
+  } catch (error) {
+    yield put({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: 'ERROR', operation: 'extract' }
+    });
+  }
+}
+export function* watchExtract() {
+  yield takeLeading('DATA_FILES_EXTRACT', extractFiles);
+}
+
+/**
+ * Create JSON string of job params
+ * @async
+ * @param {Array<Object>} files
+ * @param {String} zipfileName
+ * @returns {String}
+ */
+const getCompressParams = (files, zipfileName, latestZippy) => {
+  const inputs = {
+    inputFiles: files.map(file => `agave://${file.system}${file.path}`)
+  };
+  const parameters = {
```

Comment:
Not necessarily for now, but zippy can be refactored to not use parameters. By setting the agave application definition's inputFiles to enquote: true, you can bypass using the parameters method of passing filenames. This should allow us to replace zippy with a universal compress application that accepts multiple inputFiles (assuming that the applications submission form supports inputFiles with maxCardinality -1)

app.json:

```
"inputs": [
            {
                "id": "inputFiles",
                "value": {
                    "validator": "",
                    "visible": true,
                    "required": true,
                    "order": 0,
                    "enquote": true,
                    "default": [
                        ""
                    ]
                },
                "details": {
                    "label": "File or directory to be compressed",
                    "description": "The files or directories to be compressed.",
                    "argument": null,
                    "showArgument": false,
                    "repeatArgument": false
                },
                "semantics": {
                    "minCardinality": 1,
                    "maxCardinality": -1,
                    "ontology": [
                        "xsd:string"
                    ],
                    "fileTypes": [
                        "raw-0"
                    ]
                }
            }
        ]
```

wrapper.sh:

```
zip -rm ${outputFile} ${inputFiles}
```

Reply:
Perhaps for when we refactor compress and extract to run directly on a system where the files reside...